ovsctl/ovsctl.go

```diff
@@ -55,7 +63,26 @@ type Ovsctl struct {
 }
 
 func NewOvsctl() Ovsctl {
-	return Ovsctl{execcli: platform.NewExecClient()}
+	execcli := platform.NewExecClient()
+	// try to ensure ovs-vswitchd is active (for Ubuntu 18), otherwise log failure to start the daemon
+	if output, err := execcli.ExecuteCommand("cat /etc/os-release"); err == nil {
+		if capture := osRE.FindStringSubmatch(output); len(capture) == 2 {
+			if fields := strings.Fields(capture[1]); fields[0] == "Ubuntu" {
+				versionTokens := strings.Split(fields[1], ".")
+				if majorVersion, _ := strconv.Atoi(versionTokens[0]); majorVersion >= defaultUbuntuMajorVerion {
+					ctx := context.Background()
+					if conn, err := dbus.NewSystemdConnectionContext(ctx); err == nil {
+						defer conn.Close()
+
+						if _, err := conn.StartUnitContext(ctx, "ovs-vswitchd.service", "fail", nil); err != nil {
+							log.Printf("[ovs] Failed to start ovs-vswitchd err:%v", err)
+						}
+					}
+				}
+			}
+		}
+	}
+	return Ovsctl{execcli: execcli}
```

Comment:
can you add field with started as true so that cni would not try get os version and start ovs service for each invocation of this api